datapackage_pipelines_measure/processors/add_github_resource.py

```diff
@@ -12,25 +13,73 @@
 
 parameters, datapackage, res_iter = ingest()
 
+# 30 authenticated requests per minute, so wait 3 secs (or use
+# GITHUB_REQUEST_WAIT_INTERVAL env var) before each request
+# (https://developer.github.com/v3/search/#rate-limit)
+REQUEST_WAIT_INTERVAL = int(settings.get('GITHUB_REQUEST_WAIT_INTERVAL', 3))
+
+
+def _make_github_request(url):
+    try:
+        headers = {
+            'Authorization': 'token {}'.format(settings['GITHUB_API_TOKEN'])
+        }
+        response = requests.get(url, headers=headers)
+        json_response = response.json()
+    except simplejson.scanner.JSONDecodeError:
+        log.error('Expected JSON in response from: {}'.format(url))
+        raise
+
+    if response.status_code != 200:
+        log.error('Response from Github not successful')
+        raise RuntimeError(json_response)
```

Comment:
Instead of checking the response here, you could call `response.raise_for_status()` after the `response = requests.get()` line. This will also fix a bug where you're calling `response.json()` before checking if it was successsful.

Reply:
If I use `raise_for_status`, I can't pass the Github json response as an error messages. These are especially useful for diagnosing Github issues (such as rate limit messages). I've left it as is, and added a test.

Reply:
You're right.

The bug I saw was that if the request wasn't successful (e.g. 500), and didn't return a JSON, you'd call `response.json()`, which would fail the decoding because the response isn't a JSON. However, as you're always calling the GitHub API, it should always return a JSON, regardless of the status code, so this won't happen in practice. If that wasn't the case, we could call `raise_for_status`, and catch the `HTTPError` thrown to log the `response.text` before re-throwing the exception.